sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Expression.scala

```diff
@@ -86,12 +86,20 @@ abstract class Expression extends TreeNode[Expression] {
       case (i1, i2) => i1 == i2
     }
   }
+
+  /**
+   * Check the input data types, returns `TypeCheckResult.success` if it's valid,
+   * or return a `TypeCheckResult` with an error message if invalid.
+   * TODO: we should remove the default implementation and implement it for all
+   * expressions with proper error message.
+   */
+  def checkInputDataTypes(): TypeCheckResult = TypeCheckResult.success
```

Comment:
talked to @marmbrus -- since it is named with a "verb", and might be expensive, let's keep the parentheses here for now.